x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/hooks/use_create_analytics_form/actions.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FormMessage, State } from './state';
+
+export enum ACTION {
+  ADD_REQUEST_MESSAGE,
+  RESET_REQUEST_MESSAGES,
+  CLOSE_MODAL,
+  OPEN_MODAL,
+  RESET_ADVANCED_EDITOR_MESSAGES,
+  RESET_FORM,
+  SET_ADVANCED_EDITOR_STR,
```

Comment:
`SET_ADVANCED_EDITOR_STR` - is this for setting the complete config string? Maybe make this action name clearer.